projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts

```diff
@@ -46,31 +46,48 @@ export class AcaMobileAppSwitcherService {
     this.mobileAppSwitchConfig = this.config.get<MobileAppSwitchConfigurationOptions>('mobileAppSwitch');
   }
 
-  checkForMobileApp(): void {
-    const currentTime: number = new Date().getTime();
-    const sessionTime: string = sessionStorage.getItem('mobile_notification_expires_in');
+  resolveExistenceOfDialog(): void {
+    const url: string = this.getCurrentUrl();
+    const queryParams: string = url.split('?')[1];
+    let queryParamsList = [];
+    let hideBanner = false;
+    if (queryParams !== null && queryParams !== undefined) {
+      queryParamsList = queryParams.split('&');
+      queryParamsList.forEach((param) => {
```

Comment:
You can replace forEach with 'some' function from array probably

Reply:
we are just iterating throught the array and retrieving the queryParams for checking only and not working on the result of the array so can't it be fine

Reply:
well you can do something like: 
```ts
hideBanner = queryParamsList.some((param) => {
  const key = param.split('=')[0];
  const value = param.split('=')[1];
  return key === 'mobileapps' && value === 'true'; 
});
```

It wold be better because is more descriptive that you are looking for boolean and also loop will finish when at least one will be true instead of like for forEach - going till end of array.

Reply:
You can even make it shorter like: 
```ts
hideBanner = queryParamsList.some((param) => return param.split('=')[0] === 'mobileapps' && param.split('=')[1] === 'true');
```

Reply:
I now changed to a more shorter form as you said